validator/sawtooth_validator/execution/processor_iterator.py

```diff
@@ -63,7 +67,13 @@ def get_next_of_type(self, processor_type):
         """
         with self._condition:
             if processor_type in self:
-                return self[processor_type].next_processor()
+                for _ in range(len(self[processor_type])):
+                    processor = self[processor_type].next_processor()
+                    occupancy = self.occupancy_map.get(
+                        processor.connection_id)
+                    if occupancy['maximum'] > occupancy['current']:
+                        return processor
+                raise ProcessorQueuesFullError()
```

Comment:
This function has kind of a funky interface. It returns None if there are no processors of the given type registered but raises an exception if they are all full. Then when it is used, you aren't checking for None and are checking whether the processor is registered manually (which theoretically could change in between the line you check and the line you call this function). Instead, I would create a new error type for no processor registered and catch that where you call this function.

Reply:
This is where I am talking about adding another catch https://github.com/hyperledger/sawtooth-core/pull/1456/commits/5dbe78a9c39e69639685a4bf00b315f0a5f96b57#diff-6e2f0b2b49825b88875126268a6cff5fR340